### DISCUSSION TITLE
MySQL 8 authentication protocol problem

### DISCUSSION THREAD

**Discussion by sajaddp** [adonisjs/core#2941] [Help] [answered]
Hi,
I using Mysql 8 and followed the instructions in https://docs.adonisjs.com/guides/database/introduction.
But I got an error with this message:
```
ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; consider upgrading MySQL client
```
What is your suggestion to solve this issue?

**Comment by sajaddp** [comment, marked as answer]
**Solution**:
Install [mysql2 ](https://www.npmjs.com/package/mysql2) with ` npm i -S mysql2` and in `database.ts` change `client: 'mysql'` to `client: 'mysql2'`

**Reply by matondomp** [reply]
it was also my problams, i used this solution and  works well

**Reply by feliperafaelbarbosa** [reply]
Thanks, this worked for me. :smile:
